kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -56,9 +56,16 @@ object KinesisSink {
     val clients = for {
       credentials <- getProvider(kinesisConfig.aws)
       kinesisClient = createKinesisClient(credentials, kinesisConfig.endpoint, kinesisConfig.region)
-      _ <- if (streamExists(kinesisClient, streamName)) true.asRight
-      else new IllegalArgumentException(s"Kinesis stream $streamName doesn't exist").asLeft
       sqsClientAndName <- sqsBuffer(sqsBufferName, credentials, kinesisConfig.region)
+      _ <- if (streamExists(kinesisClient, streamName)) {
+        true.asRight
+      } else if (sqsClientAndName.isDefined) {
```

Comment:
Shouldn't we also check if it really exists and not just defined?

Reply:
Good point, I've added that check.